lib/internal/modules/helpers.js

```diff
@@ -372,7 +373,7 @@ function stripTypeScriptTypes(source, filename) {
     sourceMap: sourceMapEnabled,
     filename,
   };
-  const { code, map } = parse(source, options);
+  const { code, map } = tryParse(() => parse(source, options));
```

Comment:
Why pass a function? It seems to be pure overhead
```suggestion
  const { code, map } = tryParse(source, options);
```

Reply:
the parse function is a variable

Reply:
Yeah, let's simply move the variable declaration to the correct function then

Reply:
it requires a bit of refactor, on it

Reply:
I don't see how it's related, it would still be called, except after the `options` declaration, but that shouldn't be observable.

Reply:
`options` uses an external variable `typeScriptParsingMode ` that is loaded by `loadTypeScriptParser ` so `loadTypeScriptParser` needs to be invoked before creating options. With the refactor its much cleaner approach